package.json

```diff
@@ -30,7 +30,7 @@
   },
   "prepublish": "npm run prepare",
   "scripts": {
-    "test": "yarn run lint && yarn extract_abi && yarn generate && yarn tsc && npm run test-js",
+    "test": "yarn run lint && yarn run truffle-compile && yarn extract_abi && yarn generate && yarn tsc && npm run test-js",
```

Comment:
1. `yarn generate` already compiles.
2. If you're at it, then it's weird that we have both `yarn run` and `npm run`, even in the same line.